include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -0,0 +1,402 @@
+							}
+						}
+
+						std::for_each(current, current + currentChannelCount,
+							[&](const decodeType& itrValue)
+							{
+								uint8_t offset = &itrValue - current;
+
+								if (maxDecodeValues[offset] > itrValue)
+									maxDecodeValues[offset] = itrValue;
+
+								if (minDecodeValues[offset] < itrValue)
+									minDecodeValues[offset] = itrValue;
+							}
+						);
+					};
+
+					{
+						core::vector3du32_SIMD localCoord;
+						for (auto& z = localCoord[2] = 0u; z < state->extent.depth; ++z)
+							for (auto& y = localCoord[1] = 0u; y < state->extent.height; ++y)
+								for (auto& x = localCoord[0] = 0u; x < state->extent.width; ++x)
+									sum(core::vectorSIMDu32(x, y, z));
+					}
+
+					auto normalizeScratch = [&](bool isNormalizedFormat, bool isSignedFormat)
+					{
+						auto normalizeAsUsual = [&](decodeType* entryScratchAdress)
+						{
+							for (uint8_t channel = 0; channel < currentChannelCount; ++channel)
+								entryScratchAdress[channel] /= maxDecodeValues[channel];
+						};
+
+						auto normalizeAsSigned = [&](decodeType* entryScratchAdress)
+						{
+							for (uint8_t channel = 0; channel < currentChannelCount; ++channel)
+								entryScratchAdress[channel] = (2.0 * entryScratchAdress[channel] - maxDecodeValues[channel] - minDecodeValues[channel]) / (maxDecodeValues[channel] - minDecodeValues[channel]);
+						};
+
+						auto normalizeAsUnsigned = [&](decodeType* entryScratchAdress)
+						{
+							for (uint8_t channel = 0; channel < currentChannelCount; ++channel)
+								entryScratchAdress[channel] = (entryScratchAdress[channel] - minDecodeValues[channel]) / (maxDecodeValues[channel] - minDecodeValues[channel]);
+						};
+
+						std::function<void(decodeType*)> normalize;
+						{
+							if (isNormalizedFormat)
+								if (isSignedFormat)
+									normalize = normalizeAsSigned;
+								else
+									normalize = normalizeAsUnsigned;
+							else
+								normalize = normalizeAsUsual;
+						}
+
+						core::vector3du32_SIMD localCoord;
+							for (auto& z = localCoord[2] = 0u; z < state->extent.depth; ++z)
+								for (auto& y = localCoord[1] = 0u; y < state->extent.height; ++y)
+									for (auto& x = localCoord[0] = 0u; x < state->extent.width; ++x)
+									{
+										const size_t scratchOffset = asset::IImage::SBufferCopy::getLocalByteOffset(localCoord, scratchByteStrides);
+										decodeType* entryScratchAdress = (decodeType*)(reinterpret_cast<uint8_t*>(scratchMemory) + scratchOffset);
+
+										normalize(entryScratchAdress);
+									}
+					};
+					
+					bool normalized = asset::isNormalizedFormat(inFormat);
+					if (state->normalizeImageByTotalSATValues || normalized)
+						normalizeScratch(normalized, asset::isSignedFormat(inFormat));
+
+					{
+						const auto blockDims = asset::getBlockDimensions(outFormat);
```

Comment:
output format must always be not block-compressed (because we dont support `encodePixels` for block-compressed formats) so this is not needed

Reply:
https://github.com/buildaworldnet/IrrlichtBAW/pull/497#discussion_r470679905

Reply:
no need for you to know block dimensions for the `outFormat`